aspnet/fundamentals/localization.rst

```diff
@@ -0,0 +1,337 @@
+		app.UseIISPlatformHandler();
+
+		var options = new RequestLocalizationOptions();
+		options.RequestCultureProviders.Clear();
+		options.RequestCultureProviders.Add(new AcceptLanguageHeaderRequestCultureProvider() );
+		app.UseRequestLocalization(options);
+
+		...
+	}
+
+Implementing a custom IRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As you've seen, ASP.NET defines culture providers that attempt to establish the culture based on the Accept-Language value, a value in the query string, or a cookie. However, there might be cases not covered by these built-in providers. For example, let's say you have an app that saves user settings - such as language - on a server so that when the user logs in to your app from any machine, the user always enjoys the same experience. This is accomplished by defining your own class that implements the ``IRequestCultureProvider`` interface.
+
+In the following code snippet, I've defined a class called ``MyRequestCultureProvider`` that returns the Spanish culture. In a real-world app, you would insert your own app-specific logic here - such as reading the culture from a database - to determine the correct culture for the user.
+
+.. code-block:: c#
+	:emphasize-lines: 3-11,21-24
+
+	public class MyRequestCultureProvider : IRequestCultureProvider
+	{
+		public Task<RequestCulture> DetermineRequestCulture(HttpContext httpContext)
+		{
+			// Replace the following with your own logic to determine what
+			// culture should be used. For example, you could read the culture
+			// information from a database keyed by the logged-in user.
+			var culture = new CultureInfo("es-ES");
+			var uiCulture = new CultureInfo("en-ES");
+			return Task.FromResult(new RequestCulture(culture, uiCulture));
+		}
+	}
+
+	public class Startup
+	{
+		public void Configure(IApplicationBuilder app, IStringLocalizer<Startup> SR)
+		{
+			// Add the platform handler to the request pipeline.
+			app.UseIISPlatformHandler();
+
+			var options = new RequestLocalizationOptions();
+			options.RequestCultureProviders.Clear();
+			options.RequestCultureProviders.Add(new MyRequestCultureProvider() );
+			app.UseRequestLocalization(options);
+
+			...
+		}
+	}
+
+Accessing localized content with IStringLocalizer
+-------------------------------------------------
+
+Once the request localization middleware establishes the current culture, the ``IStringLocalizer`` service then provides access to localized content based on the current culture. To enable support for these localization services, you call the ``IServiceCollection.AddLocalization`` extension method specifying a setup action to configure the service. The most common way to call ``IServiceCollection.AddLocalization`` is as follows where the ``ResourcePath`` specifies the path where the localized resources are located. In this example, the resource files reside in a directory named `My/Resources` relative to the application root.
```

Comment:
I think technically IStringLocalizer isn't a service because you can't resolve it directly from DI. Instead you can resolve either an `IStringLocalizerFactory` or (for convenience) an `IStringLocalizer<T>`.